e2e-tests/src/test/staking.rs

```diff
@@ -195,3 +102,126 @@ pub fn staking_test(config: &Config) -> anyhow::Result<()> {
 
```

Comment:
Maybe remove the above sanity check, as it would be better suited for your test, but maybe even better would be just to move finalization test in main.rs as the last one?

Reply:
Done